pkg/azuredisk/azuredisk.go

```diff
@@ -145,6 +145,7 @@ func newDriverV1(options *DriverOptions) *Driver {
 	driver.Version = driverVersion
 	driver.NodeID = options.NodeID
 	driver.VolumeAttachLimit = options.VolumeAttachLimit
+	driver.ReservedDataDiskSlotNum = options.ReservedDataDiskSlotNum
```

Comment:
it seems this is a bug, `ReservedDataDiskSlotNum` never takes effect in the before?

Reply:
can you make this change as a separate PR? I will cherrypick to old releases, thx

Reply:
done in !2384

Reply:
No it doesn't after testing the CSIDriver object is never modified